imagedephi/rules.py

```diff
@@ -0,0 +1,94 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+from enum import Enum
+from typing import TYPE_CHECKING
+
+import tifftools
+
+if TYPE_CHECKING:
+    from tifftools.tifftools import IFD, TagEntry
+
+
+@dataclass
+class TiffTagEntry:
+    tag: tifftools.TiffTag
+    entry: TagEntry
+
+
+class RuleType(Enum):
+    METADATA = "metadata"
+    IMAGE = "image"
+
+
+class RedactMethod(Enum):
+    REPLACE = "replace"
+    DELETE = "delete"
+    KEEP = "keep"
+
+
+class RuleFormat(Enum):
+    TIFF = "tiff"
+
+
+class RuleSource(Enum):
+    BASE = "base"
+    OVERRIDE = "override"
+
+
+@dataclass
+class Rule:
+    title: str
```

Comment:
I'd suggest making `title` optional, and actually naming it `description`. In most cases, I believe we have enough information from the rest of the rule to auto-generate the printed rule description (which an explcit `description` value would override). This would typically make rules more maintainable and less likely to contain contradictory content, since I suspect a lot of people will be copy-pasting them.

Reply:
Agreed. See 676431c